cmd/karavictl/cmd/inject.go

```diff
@@ -1193,7 +1193,11 @@ func getSecretData(s *corev1.Secret) ([]SecretData, error) {
 	var ret []SecretData
 	err := json.NewDecoder(bytes.NewReader(data)).Decode(&ret)
 	if err != nil {
-		return nil, fmt.Errorf("decoding secret data json: %w", err)
+		// Got an error with JSON decode, try to decode as YAML
+		yamlErr := yaml.Unmarshal(data, &ret)
+		if yamlErr != nil {
+			return nil, fmt.Errorf("decoding secret data: %w", yamlErr)
```

Comment:
Can the json error be returned too? In the case where the secret is malformed json the returned error might be something about yaml which will be misleading.

Maybe something like `fmt.Errorf("decoding secret data: yaml error: %w, json error: %w", yamlErr, err)`

Reply:
Thanks, Aaron. I've updated, but not that I'm now using '%v' for the formatter. 

TIL, '%w' is only allowed once in an error format-- https://golang.org/src/fmt/errors.go?s=624:674#L7 :
> // If the format specifier includes a %w verb with an error operand,
> // the returned error will implement an Unwrap method returning the operand. It is
> // invalid to include more than one %w verb or to supply it with an operand
> // that does not implement the error interface. The %w verb is otherwise
> // a synonym for %v.

Reply:
Maybe they can be combined somehow. Not sure what is the best practice.

```
combinedErr:= fmt.Errorf("json: %s, yaml: %s", err.Error(), yamlErr.Error())

fmt.Errorf("decoding secret data: %w", combinedErr)
```

Reply:
It looks like using %v twice works. This is the output:

```
    inject_test.go:191: getting secret data: decoding secret data: yaml error: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type []cmd.SecretData, json error: invalid character 'T' looking for beginning of value
    --- FAIL: TestListChangePowerFlexWithBadSecretConfig/inject_a_new_secret_with_localhost_endpoints (0.01s)
```